hw/usb_interface/ch552_fw/encode_usb_strings.py

```diff
@@ -20,37 +20,51 @@ def descriptor_to_string(descriptor):
 def string_to_descriptor(string):
     """ Convert a python string into a bytes object containing a USB string descriptor"""
     descriptor = bytearray()
-    descriptor.append(0x00) # placeholder for length
-    descriptor.append(0x03)
+    descriptor.append(0x00) # Placeholder for length
+    descriptor.append(0x03) # Descriptor type (String)
     descriptor.extend(string.encode('utf-16')[2:]) # crop the BOM
-    descriptor[0] = len(descriptor)
+    descriptor[0] = len(descriptor) # Set length of this descriptor (in bytes)
 
     return bytes(descriptor)
 
+def format_descriptor(name, value):
+    descriptor = string_to_descriptor(value)
+    formatted = [
+        'unsigned char FLASH {}[] = {{  // "{}"'.format(name, value),  # Add string as a comment
+        '    {},    // Length of this descriptor (in bytes)'.format(descriptor[0]),
+        '    0x03,  // Descriptor type (String)'
+    ]
+
+    formatted.extend(
+        [
+            '    ' + ', '.join(
+                ["'{}', 0".format(chr(b)) if b != 0x00 else "0x00" for b in descriptor[2 + i:2 + i + 8:2]]
```

Comment:
Does this remove UTF-16 support? Since every other byte is replaced with a `0`.

I guess it'll work since we only use ASCII characters.